Content/ContentTypeResolver/SingleCategorySelectionResolver.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\HeadlessBundle\Content\ContentTypeResolver;
+
+use JMS\Serializer\SerializationContext;
+use Sulu\Bundle\CategoryBundle\Category\CategoryManagerInterface;
+use Sulu\Bundle\HeadlessBundle\Content\ContentView;
+use Sulu\Bundle\HeadlessBundle\Content\Serializer\CategorySerializerInterface;
+use Sulu\Component\Content\Compat\PropertyInterface;
+
+class SingleCategorySelectionResolver implements ContentTypeResolverInterface
+{
+    public static function getContentType(): string
+    {
+        return 'single_category_selection';
+    }
+
+    /**
+     * @var CategoryManagerInterface
+     */
+    private $categoryManager;
+
+    /**
+     * @var CategorySerializerInterface
+     */
+    private $categorySerializer;
+
+    public function __construct(
+        CategoryManagerInterface $categoryManager,
+        CategorySerializerInterface $categorySerializer
+    ) {
+        $this->categoryManager = $categoryManager;
+        $this->categorySerializer = $categorySerializer;
+    }
+
+    public function resolve($data, PropertyInterface $property, string $locale, array $attributes = []): ContentView
+    {
+        if (null === $data) {
+            return new ContentView([]);
```

Comment:
Think a single selection should return null instead of an empty array when not set /cc @nnatter

Reply:
yes, would use `null` here to be consistent with the existing single-selection resolvers 🙂  see for example https://github.com/sulu/SuluHeadlessBundle/blob/master/Content/ContentTypeResolver/SinglePageSelectionResolver.php

Reply:
Your right :) 
Change it.